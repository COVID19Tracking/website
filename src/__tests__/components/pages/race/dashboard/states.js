import React from 'react'
import renderer from 'react-test-renderer'
import { useStaticQuery } from 'gatsby'
import States from '~components/pages/race/dashboard/states'

beforeEach(() => {
  useStaticQuery.mockImplementation(() => ({
    allCovidRaceDataCombined: {
      nodes: [
        {
          name: 'AZ',
          state: 'Arizona',
        },
        {
          name: 'DC',
          state: 'Washington, DC',
        },
      ],
    },
    allCovidStateInfo: {
      nodes: [
        {
          childSlug: {
            slug: 'alaska',
          },
          state: 'AK',
        },
        {
          childSlug: {
            slug: 'alabama',
          },
          state: 'AL',
        },
        {
          childSlug: {
            slug: 'arkansas',
          },
          state: 'AR',
        },
      ],
    },
    allCovidRaceDataSeparate: {
      nodes: [
        {
          name: 'WA',
          state: 'Washington',
        },
        {
          name: 'GA',
          state: 'Georgia',
        },
      ],
    },
    disparityNote: {
      contentful_id: 'aNeWRaNdoMSTrinG',
      childContentfulSnippetContentTextNode: {
        childMarkdownRemark: {
          html: '<h2>Content</h2><p>More disparity snippet content</p>',
        },
      },
    },
    comparibleNote: {
      contentful_id: 'RaNdoMSTrinG',
      childContentfulSnippetContentTextNode: {
        childMarkdownRemark: {
          html: '<h2>Content</h2><p>More comparible snippet content</p>',
        },
      },
    },
  }))
})

describe('Components : Pages : Race : Dashboard : States', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<States />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
