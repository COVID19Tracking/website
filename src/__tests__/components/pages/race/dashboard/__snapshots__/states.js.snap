// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components : Pages : Race : Dashboard : States renders correctly 1`] = `
<section>
  <div
    className="state"
  >
    <h2
      className="header"
      id="state-arizona"
    >
      AZ
    </h2>
    <p
      className="socialCardCta"
    >
      Compare 
      AZ
      ’s infection and mortality data by race and ethnicity.
      <br />
      <a
        className="cta"
        href="/race/infection-and-mortality-data#Arizona"
      >
        View data per capita as a graphic
        <span
          aria-hidden={true}
          className="arrow"
        >
           
          →
        </span>
      </a>
    </p>
    <div>
      <div>
        <div
          className="stateOverview"
        >
          <div
            className="note"
          >
            <div
              className="container"
            >
              <p>
                The following tables reflect only those cases and deaths where
                 
                race/ethnicity
                 is known and reported by 
                AZ
                . If this state’s reporting percentages are low, interpret with caution.
              </p>
            </div>
          </div>
        </div>
        <h3
          className="tableTitle"
        >
          Cases and deaths by race/ethnicity
        </h3>
        <div>
          <table
            className="table"
          >
            <thead>
              <tr>
                <th
                  className="group"
                  scope="col"
                >
                  <span
                    role="text"
                  >
                    Race/ethnicity
                  </span>
                </th>
                <th
                  className="borderLeft percent"
                  scope="col"
                >
                  <span
                    role="text"
                  >
                    Percentage of population
                  </span>
                </th>
                <th
                  className="percent missingData"
                  scope="col"
                >
                  <span
                    role="text"
                  >
                    <img
                      alt="Alert icon"
                      src="test-file-stub"
                    />
                    Percentage of cases
                    <div
                      className="symbolSpacer"
                    />
                  </span>
                </th>
                <th
                  className="percent missingData"
                  scope="col"
                >
                  <span
                    role="text"
                  >
                    <img
                      alt="Alert icon"
                      src="test-file-stub"
                    />
                    Percentage of deaths
                    <div
                      className="symbolSpacer"
                    />
                  </span>
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <th
                  className="group"
                  scope="row"
                >
                  <span
                    role="text"
                  >
                    Black or African American alone
                  </span>
                </th>
                <td
                  className="borderLeft"
                >
                  0
                  %
                </td>
                <td
                  className="missingData"
                  rowSpan={8}
                >
                  <span>
                    AZ
                     does not report 
                    race/ethnicity
                     data for 
                    cases
                    .
                     
                    <a
                      href="/race/get-better-data"
                    >
                      Help us get better data.
                    </a>
                  </span>
                </td>
                <td
                  className="missingData"
                  rowSpan={8}
                >
                  <span>
                    AZ
                     does not report 
                    race/ethnicity
                     data for 
                    deaths
                    .
                     
                    <a
                      href="/race/get-better-data"
                    >
                      Help us get better data.
                    </a>
                  </span>
                </td>
              </tr>
              <tr>
                <th
                  className="group"
                  scope="row"
                >
                  <span
                    role="text"
                  >
                    Hispanic or Latino *
                  </span>
                </th>
                <td
                  className="borderLeft"
                >
                  0
                  %
                </td>
              </tr>
              <tr>
                <th
                  className="group"
                  scope="row"
                >
                  <span
                    role="text"
                  >
                    Asian alone
                  </span>
                </th>
                <td
                  className="borderLeft"
                >
                  0
                  %
                </td>
              </tr>
              <tr>
                <th
                  className="group"
                  scope="row"
                >
                  <span
                    role="text"
                  >
                    Native Hawaiian and Pacific Islander alone
                  </span>
                </th>
                <td
                  className="borderLeft"
                >
                  0
                  %
                </td>
              </tr>
              <tr>
                <th
                  className="group"
                  scope="row"
                >
                  <span
                    role="text"
                  >
                    American Indian or Alaska Native alone
                  </span>
                </th>
                <td
                  className="borderLeft"
                >
                  0
                  %
                </td>
              </tr>
              <tr>
                <th
                  className="group"
                  scope="row"
                >
                  <span
                    role="text"
                  >
                    Two or more races
                  </span>
                </th>
                <td
                  className="borderLeft"
                >
                  0
                  %
                </td>
              </tr>
              <tr>
                <th
                  className="group"
                  scope="row"
                >
                  <span
                    role="text"
                  >
                    White alone
                  </span>
                </th>
                <td
                  className="borderLeft"
                >
                  0
                  %
                </td>
              </tr>
              <tr>
                <th
                  className="group"
                  scope="row"
                >
                  <span
                    role="text"
                  >
                    Some other race alone
                  </span>
                </th>
                <td
                  className="borderLeft"
                >
                  0
                  %
                </td>
              </tr>
            </tbody>
          </table>
        </div>
        <h4
          className="header"
          id="notes-arizona"
        >
          Notes
           
          <span
            className="a11y-only"
          >
             for 
            AZ
          </span>
        </h4>
        <ul
          className="list"
        >
          <li>
            <span
              className="emptyNote"
            >
              *
            </span>
             Hispanic or Latino ethnicity, any race. All other race categories in this table are defined as Not Hispanic or Latino.
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div
    className="state"
  >
    <h2
      className="header"
      id="state-washington, dc"
    >
      DC
    </h2>
    <p
      className="socialCardCta"
    >
      Compare 
      DC
      ’s infection and mortality data by race and ethnicity.
      <br />
      <a
        className="cta"
        href="/race/infection-and-mortality-data#Washington, DC"
      >
        View data per capita as a graphic
        <span
          aria-hidden={true}
          className="arrow"
        >
           
          →
        </span>
      </a>
    </p>
    <div>
      <div>
        <div
          className="stateOverview"
        >
          <div
            className="note"
          >
            <div
              className="container"
            >
              <p>
                The following tables reflect only those cases and deaths where
                 
                race/ethnicity
                 is known and reported by 
                DC
                . If this state’s reporting percentages are low, interpret with caution.
              </p>
            </div>
          </div>
        </div>
        <h3
          className="tableTitle"
        >
          Cases and deaths by race/ethnicity
        </h3>
        <div>
          <table
            className="table"
          >
            <thead>
              <tr>
                <th
                  className="group"
                  scope="col"
                >
                  <span
                    role="text"
                  >
                    Race/ethnicity
                  </span>
                </th>
                <th
                  className="borderLeft percent"
                  scope="col"
                >
                  <span
                    role="text"
                  >
                    Percentage of population
                  </span>
                </th>
                <th
                  className="percent missingData"
                  scope="col"
                >
                  <span
                    role="text"
                  >
                    <img
                      alt="Alert icon"
                      src="test-file-stub"
                    />
                    Percentage of cases
                    <div
                      className="symbolSpacer"
                    />
                  </span>
                </th>
                <th
                  className="percent missingData"
                  scope="col"
                >
                  <span
                    role="text"
                  >
                    <img
                      alt="Alert icon"
                      src="test-file-stub"
                    />
                    Percentage of deaths
                    <div
                      className="symbolSpacer"
                    />
                  </span>
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <th
                  className="group"
                  scope="row"
                >
                  <span
                    role="text"
                  >
                    Black or African American alone
                  </span>
                </th>
                <td
                  className="borderLeft"
                >
                  0
                  %
                </td>
                <td
                  className="missingData"
                  rowSpan={8}
                >
                  <span>
                    DC
                     does not report 
                    race/ethnicity
                     data for 
                    cases
                    .
                     
                    <a
                      href="/race/get-better-data"
                    >
                      Help us get better data.
                    </a>
                  </span>
                </td>
                <td
                  className="missingData"
                  rowSpan={8}
                >
                  <span>
                    DC
                     does not report 
                    race/ethnicity
                     data for 
                    deaths
                    .
                     
                    <a
                      href="/race/get-better-data"
                    >
                      Help us get better data.
                    </a>
                  </span>
                </td>
              </tr>
              <tr>
                <th
                  className="group"
                  scope="row"
                >
                  <span
                    role="text"
                  >
                    Hispanic or Latino *
                  </span>
                </th>
                <td
                  className="borderLeft"
                >
                  0
                  %
                </td>
              </tr>
              <tr>
                <th
                  className="group"
                  scope="row"
                >
                  <span
                    role="text"
                  >
                    Asian alone
                  </span>
                </th>
                <td
                  className="borderLeft"
                >
                  0
                  %
                </td>
              </tr>
              <tr>
                <th
                  className="group"
                  scope="row"
                >
                  <span
                    role="text"
                  >
                    Native Hawaiian and Pacific Islander alone
                  </span>
                </th>
                <td
                  className="borderLeft"
                >
                  0
                  %
                </td>
              </tr>
              <tr>
                <th
                  className="group"
                  scope="row"
                >
                  <span
                    role="text"
                  >
                    American Indian or Alaska Native alone
                  </span>
                </th>
                <td
                  className="borderLeft"
                >
                  0
                  %
                </td>
              </tr>
              <tr>
                <th
                  className="group"
                  scope="row"
                >
                  <span
                    role="text"
                  >
                    Two or more races
                  </span>
                </th>
                <td
                  className="borderLeft"
                >
                  0
                  %
                </td>
              </tr>
              <tr>
                <th
                  className="group"
                  scope="row"
                >
                  <span
                    role="text"
                  >
                    White alone
                  </span>
                </th>
                <td
                  className="borderLeft"
                >
                  0
                  %
                </td>
              </tr>
              <tr>
                <th
                  className="group"
                  scope="row"
                >
                  <span
                    role="text"
                  >
                    Some other race alone
                  </span>
                </th>
                <td
                  className="borderLeft"
                >
                  0
                  %
                </td>
              </tr>
            </tbody>
          </table>
        </div>
        <h4
          className="header"
          id="notes-washington, dc"
        >
          Notes
           
          <span
            className="a11y-only"
          >
             for 
            DC
          </span>
        </h4>
        <ul
          className="list"
        >
          <li>
            <span
              className="emptyNote"
            >
              *
            </span>
             Hispanic or Latino ethnicity, any race. All other race categories in this table are defined as Not Hispanic or Latino.
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div
    className="state"
  >
    <h2
      className="header"
      id="state-georgia"
    >
      GA
    </h2>
    <p
      className="socialCardCta"
    >
      Compare 
      GA
      ’s infection and mortality data by race and ethnicity.
      <br />
      <a
        className="cta"
        href="/race/infection-and-mortality-data#Georgia"
      >
        View data per capita as a graphic
        <span
          aria-hidden={true}
          className="arrow"
        >
           
          →
        </span>
      </a>
    </p>
    <div>
      <div
        className="container"
      >
        <table
          aria-hidden={true}
          className="table"
          role="presentation"
        >
          <tbody>
            <tr>
              <td />
              <td />
              <td />
              <td />
            </tr>
            <tr>
              <td />
              <td />
              <td />
              <td />
            </tr>
            <tr>
              <td />
              <td />
              <td />
              <td />
            </tr>
            <tr>
              <td />
              <td />
              <td />
              <td />
            </tr>
            <tr>
              <td />
              <td />
              <td />
              <td />
            </tr>
            <tr>
              <td />
              <td />
              <td />
              <td />
            </tr>
          </tbody>
        </table>
        <div
          className="alert"
        >
          <div
            className="title"
          >
            <img
              alt="Alert icon"
              src="test-file-stub"
            />
            <span>
              No Data
            </span>
          </div>
          <div
            className="content"
          >
            <p>
              GA
               does not share data about race or ethnicity.
               
              <a
                href="/race/get-better-data"
              >
                Help us advocate for better data
              </a>
              .
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="state"
  >
    <h2
      className="header"
      id="state-washington"
    >
      WA
    </h2>
    <p
      className="socialCardCta"
    >
      Compare 
      WA
      ’s infection and mortality data by race and ethnicity.
      <br />
      <a
        className="cta"
        href="/race/infection-and-mortality-data#Washington"
      >
        View data per capita as a graphic
        <span
          aria-hidden={true}
          className="arrow"
        >
           
          →
        </span>
      </a>
    </p>
    <div>
      <div
        className="container"
      >
        <table
          aria-hidden={true}
          className="table"
          role="presentation"
        >
          <tbody>
            <tr>
              <td />
              <td />
              <td />
              <td />
            </tr>
            <tr>
              <td />
              <td />
              <td />
              <td />
            </tr>
            <tr>
              <td />
              <td />
              <td />
              <td />
            </tr>
            <tr>
              <td />
              <td />
              <td />
              <td />
            </tr>
            <tr>
              <td />
              <td />
              <td />
              <td />
            </tr>
            <tr>
              <td />
              <td />
              <td />
              <td />
            </tr>
          </tbody>
        </table>
        <div
          className="alert"
        >
          <div
            className="title"
          >
            <img
              alt="Alert icon"
              src="test-file-stub"
            />
            <span>
              No Data
            </span>
          </div>
          <div
            className="content"
          >
            <p>
              WA
               does not share data about race or ethnicity.
               
              <a
                href="/race/get-better-data"
              >
                Help us advocate for better data
              </a>
              .
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="js-disabled"
  >
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "<h2>Content</h2><p>More disparity snippet content</p>",
        }
      }
      id="notes-disparity"
    />
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "<h2>Content</h2><p>More comparible snippet content</p>",
        }
      }
      id="notes-comparible"
    />
  </div>
</section>
`;
